src/System/Wlog/PureLogging.hs

```diff
@@ -123,3 +124,12 @@ launchNamedPureLog hoist' (NamedPureLogger action) = do
     name <- askLoggerName
     (logs, res) <- hoist' $ swap <$> usingLoggerName name (runPureLog action)
     res <$ dispatchEvents logs
+
+-- | Similar to 'runNamedPureLog', but using provided logger name.
+usingPureLoggerName
+    :: Monad m
```

Comment:
I think you can remove `Monad` constraint here and from `runPureLog` function as well. Or at least you can relax it to `Functor`...